sycl/plugins/hip/pi_hip.cpp

```diff
@@ -2131,8 +2131,27 @@ pi_result hip_piMemGetInfo(pi_mem memObj, pi_mem_info queriedInfo,
 /// \return PI_SUCCESS
 pi_result hip_piextMemGetNativeHandle(pi_mem mem,
                                       pi_native_handle *nativeHandle) {
+#if defined(__HIP_PLATFORM_NVIDIA__)
+  if (sizeof(_pi_mem::mem_::buffer_mem_::native_type) >
+      sizeof(pi_native_handle)) {
+    // Check that all the upper bits that cannot be represented by
+    // pi_native_handle are empty.
+    // NOTE: The following shift might trigger a warning, but the check in the
+    // if above makes sure that this does not underflow.
+    _pi_mem::mem_::buffer_mem_::native_type upperBits =
+        mem->mem_.buffer_mem_.get() >> (sizeof(pi_native_handle) * CHAR_BIT);
+    if (upperBits) {
+      // Return an error if any of the remaining bits is non-zero.
+      return PI_INVALID_MEM_OBJECT;
+    }
+  }
+  *nativeHandle = static_cast<pi_native_handle>(mem->mem_.buffer_mem_.get());
+#elif defined(__HIP_PLATFORM_AMD__)
```

Comment:
> can you approve the workflow again, please?

My outlook filters for github notifications aren't perfect yet and different notifications end up in different folders. Just saw your comment.

I'd like to understand what exactly is wrong with the AMD platform because it's not obvious to me. The upperbits handling looks like the right thing to me regardless of the problem.

Reply:
Just FYI. I approved the workflow both times.

Reply:
@bader Thanks for approving the workflows! I'm not sure why the second run failed, I don't see how my changes could affect a reduction test on OpenCL on Intel GPU.

@aelovikov-intel: The definition of `hipDeviceptr_t` (which is the same as `native_type`, the return of `mem->mem_buffer_mem_.get()`) is different on Nvidia and AMD platforms.

On Nvidia/CUDA platforms, it is an alias for `CUdeviceptr`, which is `unsigned long long`, so a `static_cast` is required to convert it to `pi_native_handle` (`unsigned long`).

On AMD platforms, it is an alias for `void*`, and `static_cast` of `void*` to `unsigned long` are invalid, so a `reinterpret_cast` is required on the AMD platform (and no checking of upper bits).

Reply:
And such a cast would error if information would be lost: https://godbolt.org/z/dj8TKhrnM

That makes sense, thanks!